.gitlab-ci.yml

```diff
@@ -0,0 +1,92 @@
+workflow:
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "merge_request_event"
+      auto_cancel:
+        on_new_commit: interruptible
+    - if: $CI_COMMIT_BRANCH == "main"
+      auto_cancel:
+        on_new_commit: conservative
+    - if: $CI_COMMIT_TAG
+      auto_cancel:
+        on_new_commit: conservative
+
+default:
+  interruptible: true
+  before_script:
+    # FIXME: build minimal el9 container, or use container-forge
+    - dnf install -y java-21-openjdk maven wget tree ruby python3-pip
+
+stages:
+  - build-stage
+  - test-stage
+  - deploy-stage
+
+build:
+  stage: build-stage
+  script:
+    - ./build-coatjava.sh -T4 --unittests --spotbugs --quiet --no-progress
+    - tar -czf coatjava.tar.gz coatjava
```

Comment:
You can just exclude this since the artifact `coatjava/` should preserve permissions (unlike GitHub, where we had to `tar` it).
```suggestion
```